pages/proposals/create.tsx

```diff
@@ -1,96 +1,34 @@
+import { messageForProposal } from '../../models/proposal/proposalSelectors'
 import { defaultExecuteFee } from '../../util/fee'
-import { isValidJson } from '../../util/isValidJson'
-import { makeSpendMessage } from '../../util/messagehelpers'
-
-interface FormElements extends HTMLFormControlsCollection {
-  label: HTMLInputElement
-  description: HTMLInputElement
-  json: HTMLInputElement
-}
-
-interface ProposalFormElement extends HTMLFormElement {
-  readonly elements: FormElements
-}
 
 const contractAddress = process.env.NEXT_PUBLIC_DAO_CONTRACT_ADDRESS || ''
 
 const ProposalCreate: NextPage = () => {
   const router = useRouter()
 
   const { walletAddress, signingClient } = useSigningClient()
-  const [sendWalletAddress, setSendWalletAddress] = useState('')
   const [transactionHash, setTransactionHash] = useState('')
   const [error, setError] = useState('')
   const [loading, setLoading] = useState(false)
-  const [proposalMessage, setProposalMessage] = useState<any>() // message type?
-  const [messageJson, setMessageJson] = useState('')
   const [proposalID, setProposalID] = useState('')
 
-  const handleSpend = () => {
-    const amount = prompt('Amount?')
-    if (amount) {
-      const spendMsg = makeSpendMessage(
-        amount,
-        sendWalletAddress || walletAddress
-      )
-      setProposalMessage(spendMsg)
-      setMessageJson(JSON.stringify(spendMsg))
-    }
-  }
-
-  const handleSubmit = async (event: FormEvent<ProposalFormElement>) => {
-    event.preventDefault()
+  const handleProposal = async (proposal: Proposal) => {
     setLoading(true)
     setError('')
-
-    const currentTarget = event.currentTarget as ProposalFormElement
-
-    const title = currentTarget.label.value.trim()
-    const description = currentTarget.description.value.trim()
-    const jsonStr = currentTarget.json.value.trim()
-
-    if (title.length === 0 || description.length === 0) {
-      setLoading(false)
-      setError('Title and Description are required.')
-    }
-
-    // clone json string to avoid prototype poisoning
-    // https://medium.com/intrinsic-blog/javascript-prototype-poisoning-vulnerabilities-in-the-wild-7bc15347c96
-    let json
-    const jsonClone = cloneDeep(jsonStr)
-    if (jsonClone) {
-      try {
-        json = JSON.parse(jsonClone)
-        if (!isValidJson(json)) {
-          setLoading(false)
-          setError('Error in JSON message.')
-          return
-        }
-      } catch {
-        setLoading(false)
-        setError('Proposal is not valid JSON.')
-        return
-      }
-    }
-
-    const msg = {
-      title,
-      description,
-      msgs: json || [],
-    }
-
+    const propose = messageForProposal(proposal)
+    const memo = memoForProposal(proposal)
     try {
       const response = await signingClient?.execute(
         walletAddress,
         contractAddress,
-        { propose: msg },
+        { propose, memo },
         defaultExecuteFee
```

Comment:
```suggestion
        propose,
        defaultExecuteFee,
        memo
```

This was causing an error, and is not the right way to pass in a memo: https://cosmos.github.io/cosmjs/latest/cosmwasm-stargate/classes/signingcosmwasmclient.SigningCosmWasmClient.html#execute